addons/sale/models/sale.py

```diff
@@ -637,19 +637,23 @@ def _create_invoices(self, grouped=False, final=False, date=None):
             invoice_vals = order._prepare_invoice()
 
             # Invoice line values (keep only necessary sections).
+            number_of_notes_in_invoice_vals = 0
             for line in order.order_line:
                 if line.display_type == 'line_section':
                     pending_section = line
                     continue
-                if float_is_zero(line.qty_to_invoice, precision_digits=precision):
+                if line.display_type == 'line_note':
+                    number_of_notes_in_invoice_vals += 1
+                elif float_is_zero(line.qty_to_invoice, precision_digits=precision):
                     continue
-                if line.qty_to_invoice > 0 or (line.qty_to_invoice < 0 and final):
+                if line.qty_to_invoice > 0 or (line.qty_to_invoice < 0 and final) or line.display_type == 'line_note':
+                    line.sequence += len(invoice_vals_list) * 100
                     if pending_section:
                         invoice_vals['invoice_line_ids'].append((0, 0, pending_section._prepare_invoice_line()))
                         pending_section = None
                     invoice_vals['invoice_line_ids'].append((0, 0, line._prepare_invoice_line()))
 
-            if not invoice_vals['invoice_line_ids']:
+            if not invoice_vals['invoice_line_ids'] or len(invoice_vals['invoice_line_ids']) == number_of_notes_in_invoice_vals:
```

Comment:
I would make an explicit condition instead of a counter
  * any returns False if list is void
  * we check for at least one real line (display_type is False)

Aka
```
if not any(new_line['display_type'] is False for (_, _, new_line) in invoice_vals['invoice_line_ids'])
```

We could also move invoice_line_ids in a real variable and assign it afterwards to have a more readable code.

Reply:
👍 Implemented